test/integration-tests/src/test/java/org/wso2/testgrid/test/base/Utils.java

```diff
@@ -0,0 +1,197 @@
+
+        ShellExecutor executor = new ShellExecutor();
+        return executor.executeJar(getJarPath(), cmdArgs);
+    }
+
+    /**
+     * Construct the executable Jar absolute path and returns the value.
+     *
+     * @return Path of the executable Jar
+     */
+    private static String getJarPath() {
+        // Generating the actual Jar name with the version
+        String jarPath = TG_EXECUTABLE_LOCATION.replace("*", System.getProperty("project.version"));
+        return TESTGRID_UNZIP_LOCATION + jarPath;
+    }
+
+    /**
+     * Unzip a zip file into a given location.
+     *
+     * @param sourceFilePath - zip file need to extract
+     * @param extractedDir   - destination path given file to extract
+     * @throws IntegrationTestException is thrown if a exception occurs while unzipping the file
+     */
+    private static void extractFile(String sourceFilePath, String extractedDir) throws IntegrationTestException {
+        logger.info("Zip location is set to " + sourceFilePath + " || Unzip path is set to " + extractedDir);
+        FileOutputStream fileoutputstream = null;
+
+        String fileDestination = extractedDir + File.separator;
+        byte[] buf = new byte[1024];
+        ZipInputStream zipinputstream = null;
+        ZipEntry zipentry;
+        try {
+            zipinputstream = new ZipInputStream(new FileInputStream(sourceFilePath));
+
+            zipentry = zipinputstream.getNextEntry();
+
+            while (zipentry != null) {
+                //for each entry to be extracted
+                String entryName = fileDestination + zipentry.getName();
+                entryName = entryName.replace('/', File.separatorChar);
+                entryName = entryName.replace('\\', File.separatorChar);
+                int n;
+
+                File newFile = new File(entryName);
+                boolean fileCreated = false;
+                if (zipentry.isDirectory()) {
+                    if (!newFile.exists()) {
+                        fileCreated = newFile.mkdirs();
+                    }
+                    zipentry = zipinputstream.getNextEntry();
+                    continue;
+                } else {
+                    File resourceFile = new File(entryName.substring(0, entryName.lastIndexOf(File.separator)));
+                    if (!resourceFile.exists()) {
+                        if (!resourceFile.mkdirs()) {
+                            break;
+                        }
+                    }
+                }
+
+                fileoutputstream = new FileOutputStream(entryName);
+
+                while ((n = zipinputstream.read(buf, 0, 1024)) > -1) {
+                    fileoutputstream.write(buf, 0, n);
+                }
+
+                fileoutputstream.close();
+                zipinputstream.closeEntry();
+                zipentry = zipinputstream.getNextEntry();
+            }
+            zipinputstream.close();
```

Comment:
Closing a ZipInputStream should happen in the finally block. But this type of boiler-plate code is no longer needed because Java 7's try-with-resources feature.